### PR TITLE
doc/user: improve volatility docs

### PR DESCRIPTION
**Pull request by benesch**
Iterate on some wording improvements that didn't make the deadline for
#6320.

@sploiselle @msudberg it would be great to get a review of this whole doc more thoroughly here, too. I rushed #6320 out the door to unblock the cloud product, and I'm definitely looking to slow things down here to make sure y'all are happy with the docs.

### CODE REVIEW

**Review by sploiselle — APPROVED** [review]
LGTM mod two small comments:
- "will allow you to" can always be rewritten to "lets you", which I prefer
- The references to tables at the bottom should be nominally expanded to indicate why they're volatile, i.e. "Tables are always volatile because they do not currently retain state between restarts."

File: doc/user/content/overview/volatility.md [hunk 1]
```diff
@@ -61,10 +61,10 @@ file source.
 The volatility of a view, index, or sink is determined by applying each of
 the following rules in turn:
 
-  1. If the object depends on at least one volatile object, it is volatile.
-  2. If the object depends on at least one object of unknown volatility, it has
+  1. If the object depends on at least one volatile source, it is volatile.
+  2. If the object depends on at least one source of unknown volatility, it has
      unknown volatility.
-  3. Otherwise the object is nonvolatile.
```
Comment (mjibson) [inline doc/user/content/overview/volatility.md:67]: I think this should stay as object?

Comment (benesch) [reply doc/user/content/overview/volatility.md:67]: D'oh, yes, thanks.

**Comment by benesch** [comment]
> "will allow you to" can always be rewritten to "lets you", which I prefer

Well, except in this case, I think? 😄 That transformation changes the tense from future to present but the sentence is intentionally talking about future behavior! I changed "will allow you to" to "will let you" though, in case that is what you meant!

**Review by msudberg — APPROVED** [review]
lgtm

**Pull request merged**